src/tests/system/tests/test_memcache.py

```diff
@@ -2,1659 +2,427 @@
     client.sssd.start()
 
-    r = client.host.conn.exec(["ls", "/var/lib/sss/mc"])
-    assert r.stdout == "", "Cache directory is not empty"
-    assert r.stderr == "", "Ls command failed"
-
-    result = client.tools.id("user1")
-    assert result is not None, "User user1 was not found using user1"
-    assert result.user.id == 123456, f"User id {result.user.id} is incorrect, 123456 expected"
-
-    gresult = client.tools.getent.group("group1")
-    assert gresult is not None, "Group group1 is not found using getent"
-    assert gresult.gid == 10001, f"Group gid {gresult.gid} is incorrect, 10001 expected"
-
-    client.sssd.stop()
-
-    assert client.tools.id("user1") is None, "User user1 was found which is not expected"
-    assert client.tools.id(123456) is None, "User with id 123456 was found which is not expected"
-    assert client.tools.getent.group("group1") is None, "Group group1 was found which is not expected"
-    assert client.tools.getent.group(10001) is None, "Group with gid 10001 was found which is not expected"
-
-
-@pytest.mark.importance("critical")
-@pytest.mark.cache
-@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
-def test_memcache__removed_cache_without_invalidation(client: Client, provider: GenericProvider):
-    """
-    :title: SSSD is stopped, cache removed then users and groups cannot be lookedup
-    :setup:
-        1. Add 'user1' to SSSD
-        2. Set user uid and gid
-        3. Add 'group1' to SSSD
-        4. Set group gid
-        5. Start SSSD
-    :steps:
-        1. Find user with id(name)
-        2. Check that user has correct uid
-        3. Find group with getent.group(name)
-        4. Check that group has correct gid
-        5. Stop SSSD
-        6. Remove cache files
-        7. Find user with id(name) and id(uid)
-        8. Find group with getent.group(name) and getent.group(gid)
-    :expectedresults:
-        1. User is found
-        2. User has correct uid
-        3. Group is found
-        4. Group has correct gid
-        5. SSSD is stopped
-        6. Cache files are removed
-        7. User is not found
-        8. Group is not found
-    :customerscenario: True
-    """
-    provider.user("user1").add(uid=123456, gid=19001)
-    provider.group("group1").add(gid=10001)
+    r1 = client.tools.id(u1.name)
+    assert r1 is not None, f"`{u1.name}` should be found"
 
-    client.sssd.domain["ldap_id_mapping"] = "false"
-    client.sssd.start()
-
-    result = client.tools.id("user1")
-    assert result is not None, "User user1 is not found using id"
-    assert result.user.id == 123456, f"User id {result.user.id} is incorrect, 123456 expected"
+    r2 = client.tools.id(u2.name)
+    assert r2 is not None, f"`{u2.name}` should be found"
 
-    gresult = client.tools.getent.group("group1")
-    assert gresult is not None, "Group group1 is not found using getent"
-    assert gresult.gid == 10001, f"Group gid {gresult.gid} is incorrect, 10001 expected"
+    if by_gid:
```

Comment:
This code branching suggests there should be (at least) two  testcases instead.

Reply:
I agree with improving the readability of the tests, but I disagree breaking this test into 2.

Something nicer would be:
```
assert gresult.gid == 10001, f"Group gid {gresult.gid} is incorrect, 10001 expected"
    if search_type["by_gid"]:
...
    if search_type["by_name"]:
```

wdyt?